### PR TITLE
[tests] fix wait_for_inv()

### PR DESCRIPTION
**Pull request by jnewbery**
Fixes bug uncovered here: https://github.com/bitcoin/bitcoin/pull/10169#discussion_r114389770

### CODE REVIEW

File: test/functional/test_framework/mininode.py [hunk 1]
```diff
@@ -1604,7 +1604,10 @@ def wait_for_getheaders(self, timeout=60):
         assert wait_until(test_function, timeout=timeout)
 
     def wait_for_inv(self, expected_inv, timeout=60):
-        test_function = lambda: self.last_message.get("inv") and self.last_message["inv"] != expected_inv
+        """ Currently only checks the first inv object in an inv message"""
```
Comment (MarcoFalke) [inline test/functional/test_framework/mininode.py]: In which case you might want to `raise NotImplementedError` if `len(expected_inv) >= 2`?

Comment (jnewbery) [inline test/functional/test_framework/mininode.py]: Great idea. Thanks. Done.

**Review by sdaftuar — APPROVED** [review]
Thanks for catching this.  Looks like this was broken in two ways, which is why the test was succeeding in the first place:
- inequality instead of equality, which only didn't cause problems because
- we were comparing a p2p message with its payload (so the inequality condition would trigger, and the wait_until would return)

utACK

**Comment by luke-jr** [comment]
Needs backport?

**Pull request merged**